src/alacris_lib/state_update.ml

```diff
@@ -54,9 +53,8 @@ let post_state_update : Revision.t -> Digest.t -> TransactionReceipt.t Lwt_exn.t
   if state_update_log then
     Logging.log "post_state_update operator_revision=%s digest=%s" (Revision.to_string operator_revision)  (Digest.to_0x digest);
   let operation = make_state_update_call digest operator_revision in
-  let value = TokenAmount.zero in
   let oper_addr = Side_chain_server_config.operator_address in
-  Ethereum_user.post_operation operation oper_addr value
+  Ethereum_user.post_operation ~operation ~sender:oper_addr ~value:None
```

Comment:
Don't we want value zero here? Actually, what is the case for value None, already?

Reply:
If we have value:None then the value is zero. This is the revelation that we had from the eth_estimateGas problem. The value was set to zero because the argument was missing.

The resolution of the code smell that you saw was to implement the value as optional. This is the logical conclusion of the code smell.

Reply:
Sometimes the right thing is to carry the option around everywhere. Sometimes the right thing is to convert it to a default value ASAP.

I don't think it even means anything to have no value. It's actually a short hand notation for zero. And  OCaml already has a notion of keyword arguments with a default value, which is probably what should be used at one point to match some Ethereum interface. In other places, though, not having an option and explicitly using `zero` is the correctly typed thing, I believe.

Reply:
I would say that ethereum allows the "value" argument to be missing. And it makes sense to repeat that in the OCaml code.
That being said, if you want I can revert the changes.

Reply:
Changes have been reverted as requested.
Code has passed the tests.

Reply:
value cannot be an option, unlike gas price or gas limit.